test/periodic.jl

```diff
@@ -0,0 +1,369 @@
+@testset "periodic" begin
+    @testset "_periodic" begin
+        @testset "$f" for f in (sin, cos)
+            # A typical 24-hour day
+            for h in 0:24
+                result = _periodic(
+                    f,
+                    ZonedDateTime(2015, 6, 1, tz"America/Winnipeg") + Hour(h),
+                    Day(1),
+                )
+                expected = f(2π / 24 * h)
+                @test result ≈ expected atol=1e-14
+            end
+
+            # A spring daylight-saving time change where the day has 23-hours
+            for h in 0:23
+                result = _periodic(
+                    f,
+                    ZonedDateTime(2015, 3, 8, tz"America/Winnipeg") + Hour(h),
+                    Day(1),
+                )
+                expected = f(2π / 23 * h)
+                @test result ≈ expected atol=1e-14
+            end
+
+            # A fall daylight-saving time change where the day has 25-hours
+            for h in 0:25
+                result = _periodic(
+                    f,
+                    ZonedDateTime(2015, 11, 1, tz"America/Winnipeg") + Hour(h),
+                    Day(1),
+                )
+                expected = f(2π / 25 * h)
+                @test result ≈ expected atol=1e-14
+            end
+        end
+
+        @testset "phase shift" begin
+            @test _periodic(sin, DateTime(1), Week(1)) == 0
+            @test _periodic(sin, DateTime(1), Week(1), Day(7)) == 0
+            @test _periodic(sin, DateTime(1), Week(1), Day(-7)) == 0
+
+            @test _periodic(sin, DateTime(2), Week(1)) ≈ 0.7818314824680298 atol=1e-14
```

Comment:
I'm fine with the test but this level of precision is probably not necessary and could well be affected by rounding errors

Reply:
It's not necessary here but I think it's important to test as high a precision as possible somewhere so that we know if it becomes less precise.